core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -100,77 +93,8 @@ protected static boolean getBooleanValueOfVariable( VariableSpace space, String
     return Objects.equals( Boolean.TRUE, ValueMetaBase.convertStringToBoolean( defaultValue ) );
   }
 
-  protected VariableSpace getSpace( ConnectionDetails connectionDetails ) {
+  @NonNull
+  protected VariableSpace getSpace( @NonNull T connectionDetails ) {
     return connectionDetails.getSpace() == null ? Variables.getADefaultVariableSpace() : connectionDetails.getSpace();
   }
-
-  @Override
-  public boolean test( @NonNull T connectionDetails, @NonNull VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
```

Comment:
Moved to `VFSConnectionManagerWrapper`.
Root path testing is not provider-specific.
Manager already has as test method that delegates to the specific provider.

Reply:
This API was introduced recently, in previous PRs, to support the new root path feature, so there is no issue with backward compatibility.